contracts/pcv/PCVGuardian.sol

```diff
@@ -0,0 +1,185 @@
+
+    /// @notice governor-or-guardian-only method to un-whitelist a pcvDesposit address to withdraw funds from
+    /// @param pcvDeposit the address to un-set as safe
+    function unsetWhitelistAddress(address pcvDeposit)
+        external
+        override
+        onlyGuardianOrGovernor
+    {
+        _unsetWhitelistAddress(pcvDeposit);
+    }
+
+    /// @notice batch version of unsetWhitelistAddress
+    /// @param _whitelistAddresses the addresses to un-whitelist
+    function unsetWhitelistAddresses(address[] calldata _whitelistAddresses)
+        external
+        override
+        onlyGuardianOrGovernor
+    {
+        // improbable to ever overflow
+        unchecked {
+            require(_whitelistAddresses.length != 0, "empty");
+            for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+                _unsetWhitelistAddress(_whitelistAddresses[i]);
+            }
+        }
+    }
+
+    /// @notice governor-or-guardian-or-pcv-guard method to withdraw funds from a pcv deposit, by calling the withdraw() method on it
+    /// @param pcvDeposit the address of the pcv deposit contract
+    /// @param amount the amount to withdraw
+    function withdrawToSafeAddress(address pcvDeposit, uint256 amount)
+        external
+        override
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.GUARDIAN,
+            TribeRoles.PCV_GUARD
+        )
+    {
+        require(
+            isWhitelistAddress(pcvDeposit),
+            "Provided address is not whitelisted"
+        );
+
+        if (pcvDeposit._paused()) {
+            pcvDeposit._unpause();
+            IPCVDeposit(pcvDeposit).withdraw(safeAddress, amount);
+            pcvDeposit._pause();
+        } else {
+            IPCVDeposit(pcvDeposit).withdraw(safeAddress, amount);
+        }
+
+        emit PCVGuardianWithdrawal(pcvDeposit, amount);
+    }
+
+    /// @notice governor-or-guardian-or-pcv-guard method to withdraw all at once funds from a pcv deposit, by calling the withdraw() method on it
+    /// @param pcvDeposit the address of the pcv deposit contract
+    function withdrawAllToSafeAddress(address pcvDeposit)
+        external
+        override
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.GUARDIAN,
+            TribeRoles.PCV_GUARD
+        )
+    {
+        require(
+            isWhitelistAddress(pcvDeposit),
+            "Provided address is not whitelisted"
+        );
+
+        uint256 amountToWithdraw = IPCVDeposit(pcvDeposit).balance();
+
+        if (pcvDeposit._paused()) {
+            pcvDeposit._unpause();
+            IPCVDeposit(pcvDeposit).withdraw(safeAddress, amountToWithdraw);
+            pcvDeposit._pause();
+        } else {
+            IPCVDeposit(pcvDeposit).withdraw(safeAddress, amountToWithdraw);
+        }
+
+        emit PCVGuardianWithdrawal(pcvDeposit, amountToWithdraw);
+    }
```

Comment:
Could either make `withdrawToSafeAddress` public and call or make an internal function shared by both methods

```suggestion
    function withdrawAllToSafeAddress(address pcvDeposit)
        external
        override
    {
    uint256 amountToWithdraw = 
    IPCVDeposit(pcvDeposit).balance();
    
    withdrawToSafeAddress(pcvDeposit, amountToWithdraw)
    }
```